internal/service/wallet.go

```diff
@@ -199,30 +212,138 @@ func (s *walletService) GetWalletTransactions(ctx context.Context, walletID stri
 
 // Update the TopUpWallet method to use the new processWalletOperation
 func (s *walletService) TopUpWallet(ctx context.Context, walletID string, req *dto.TopUpWalletRequest) (*dto.WalletResponse, error) {
+	var referenceType types.WalletTxReferenceType
+	var referenceID string
+
+	var idempotencyKey *string
+	if lo.FromPtr(req.IdempotencyKey) != "" {
+		idempotencyKey = req.IdempotencyKey
+	} else {
+		idempotencyKey = lo.ToPtr(types.GenerateUUID())
+	}
+
```

Comment:
add logic to add ref type and ref id here directly irrespective of the transaction_reason